hs-abci-sdk/src/Tendermint/SDK/Modules/Auth/Types.hs

```diff
@@ -5,27 +5,79 @@ module Tendermint.SDK.Modules.Auth.Types where
 import           Control.Lens                 (Wrapped (..), from, iso, view,
                                                (&), (.~), (^.), (^..),
                                                _Unwrapped')
+import           Data.Aeson                   as JSON
 import           Data.Bifunctor               (bimap)
 import qualified Data.ProtoLens               as P
 import           Data.Proxy                   (Proxy (..))
+import           Data.String                  (IsString (..))
 import           Data.String.Conversions      (cs)
-import           Data.Text                    (Text)
+import           Data.Text                    (Text, pack)
+import qualified Data.Text.Lazy               as TL
 import           Data.Word
 import           GHC.Generics                 (Generic)
 import           GHC.TypeLits                 (symbolVal)
 import qualified Proto.Modules.Auth           as A
 import qualified Proto.Modules.Auth_Fields    as A
+import           Proto3.Suite                 (HasDefault (..), MessageField,
+                                               Primitive (..), fieldNumber)
+import qualified Proto3.Suite.DotProto        as DotProto
+import qualified Proto3.Wire.Decode           as Decode
+import qualified Proto3.Wire.Encode           as Encode
 import           Tendermint.SDK.BaseApp       (AppError (..), IsAppError (..),
                                                IsKey (..), Queryable (..))
-import           Tendermint.SDK.Codec         (HasCodec (..))
+import           Tendermint.SDK.Codec         (HasCodec (..),
+                                               defaultSDKAesonOptions)
 import           Tendermint.SDK.Types.Address (Address)
+import           Web.HttpApiData              (FromHttpApiData (..),
+                                               ToHttpApiData (..))
 
 type AuthModule = "auth"
 
+newtype CoinId = CoinId { unCoinId :: Text } deriving (Eq, Show, Generic)
```

Comment:
I'm starting to lean toward the fact we shouldn't really use this proto-3 suite in the sdk itself, but rather reserve it as a convenience for developing apps more quickly. This is why I've included proto files for things like `Account` and `RawTransaction` in the `/protos` dir of the sdk.

It adds more boilerplate, but i think its ultimately cleaner given that protolens is a more robust library and at some point we might be taking protofiles from a central repository for cosmos modules

Reply:
Should I change this? Or perhaps make an issue after this?

Reply:
We should just do it now, it's not that bad